pkg/k8sutil/k8sutil.go

```diff
@@ -142,7 +142,8 @@ func GetPod(ctx context.Context, client crclient.Client, ns string) (*corev1.Pod
 
```

Comment:
Really thank you for contribution. IHMO it shows a great solution to filter the third-party API's and/or CRD which are cluster scoped. Following my review and lets the other inputs as well.